Dev/src/screens/StudentScreens/StudentMainScreen/StudentMainScreen.js

```diff
@@ -1,1702 +1,1511 @@
+import SideMenu from "react-native-side-menu";
+import LoadingSpinner from "components/LoadingSpinner";
+import screenStyle from "config/screenStyle";
+import fontStyles from "config/fontStyles";
+import { screenHeight, screenWidth } from "config/dimensions";
+import AudioPlayer from "components/AudioPlayer/AudioPlayer";
+import Toast, { DURATION } from 'react-native-easy-toast';
+import { LineChart } from "react-native-chart-kit";
+import CodeInput from 'react-native-confirmation-code-input';
+import DailyTracker, { getTodaysDateString } from 'components/DailyTracker';
+import themeStyles from "config/themeStyles";
+import TouchableText from "components/TouchableText";
+import QCView from "components/QCView";
+
+const translateY = new Animated.Value(-35);
+const opacity = new Animated.Value(0);
+const opacityInterpolate = opacity.interpolate({
+  inputRange: [0, 0.85, 1],
+  outputRange: [0, 0, 1]
+});
+
+const StudentMainScreen  = (props) => {
+    const [isLoading, setIsLoading] = useState(true)
+    const [student, setStudent] = useState("")
+    const [sent, setSent] = useState(null)
+    const [userID, setUserID] = useState("")
+    const [currentClass, setCurrentClass] = useState("")
+    const [currentClassID, setCurrentClassID] = useState("")
+    const [assignmentHistory, setAssignmentHistory] = useState([])
+    const [isOpen, setIsOpen] = useState(false)
+    const [studentClassInfo, setStudentClassInfo] = useState("")
+    const [modalVisible, setModalVisible] = useState(false)
+    const [recordingUIVisible, setRecordingUIVisible] = useState([])
+    const [noCurrentClass, setNoCurrentClass] = useState(false)
+    const [classCode, setClassCode] = useState("")
+    const [wordsPerAssignmentData, setWordsPerAssignmentData] = useState([])
+    const [submittedRecordings, setSubmittedRecordings] = useState(null)
+    const [classes, setClasses] = useState("")
+    const [audioFilePath, setAudioFilePath] = useState("")
+    const [isRecording, setIsRecording] = useState(false)
+    const [currentPosition, setCurrentPosition] = useState("0:00")
+    const [classesAttended, setClassesAttended] = useState(0)
+    const [classesMissed, setClassesMissed] = useState(0)
+    const [refs, setRefs] = useState({toast: useRef()})
+    const [dailyPracticeLog, setDailyPracticeLog] = useState({})
+    const [audioPlaybackVisible, setAudioPlaybackVisible] = useState(false)
+
+  //-------------- Component lifecycle methods -----------------------------------
+  //Fetches all the values for the state from the firestore database
+  useEffect(() => {
+    asyncUseEffect()
+  }, [])
+
+  const asyncUseEffect = async () => {
+
+    //Sets the screen name in firebase analytics
+    FirebaseFunctions.setCurrentScreen(
+      "Student Main Screen",
+      "StudentMainScreen"
+    );
+
+    const { userID, logAsPractice } = props.navigation.state.params;
+    const student = await FirebaseFunctions.getStudentByID(userID);
```

Comment:
All of these Functions need to be using cloud functions now. The way you call those are "FirebaseFunctions.call('getStudentByID', { studentID: userID });